src/main/java/de/tum/in/www1/artemis/service/util/structureoraclegenerator/TypesDiff.java

```diff
@@ -0,0 +1,225 @@
+	}
+
+    /**
+     * This method generates the super class diff of the solution type and the template type, e.g. the name of the class
+     * the solution type extends and the template type does not.
+     * @return The simple name of the super class the solution type extends and the template type does not. If both
+     * types are subclasses of the same class, return an empty string.
+     */
+	private String generateSuperClassName() {
+		CtTypeReference<?> solutionSuperClass = solutionType.getSuperclass();
+		CtTypeReference<?> templateSuperClass = (isTemplateNull || templateType.getSuperclass() == null) ? null : templateType.getSuperclass();
+		
+		return (solutionSuperClass != null && templateSuperClass == null) ? solutionSuperClass.getSimpleName() : "";
+	}
+
+    /**
+     * This method generates the super interfaces diff of the solution and template type, e.g. the interfaces the
+     * solution type implements, but the template type does not.
+     * @return A set of interfaces the solution type implements, but the template type does not.
+     */
+	private Set<CtTypeReference<?>> generateSuperInterfaces() {
+        // Use this predicate to filter out super interfaces that are implicit, e.g. not explicitly defined in the code.
+        Predicate<CtTypeReference<?>> interfaceIsImplicit = CtElement::isImplicit;
+
+        // Create an empty set of interfaces for the super interfaces diff and deep-copy the super interfaces of the solution type in it.
+        Set<CtTypeReference<?>> superInterfacesDiff = new HashSet<>(solutionType.getSuperInterfaces());
+		superInterfacesDiff.removeIf(interfaceIsImplicit);
+
+		// If the template is non-existent, then the super interfaces diff consists of all the super interfaces of the solution type.
+		if(!isTemplateNull) {
+
+            // Check all the super interfaces in the template type if they match to the ones in the solution type
+            // and remove them from the diff, if that's the case.
+            for(CtTypeReference<?> templateTypeSuperInterface : templateType.getSuperInterfaces()) {
+
+                // The interfaces are uniquely identified by their names.
+		        superInterfacesDiff.removeIf(solutionTypeSuperInterface ->
+                    solutionTypeSuperInterface.getSimpleName().equals(templateTypeSuperInterface.getSimpleName())
+                );
+            }
+        }
+
+		return superInterfacesDiff;
+	}
+
+    /**
+     * This method generates the methods diff of the solution and template type, e.g. the methods defined in the
+     * solution type but not in the template type.
+     * @return A set of methods defined in the solution type but not in the template type.
+     */
+    private Set<CtMethod<?>> generateMethodsDiff() {
+        // Use this predicate to filter out methods that are implicit, e.g. not explicitly defined in the code.
+        Predicate<CtMethod<?>> methodIsImplicit = m -> m.isImplicit() || m.getSimpleName().equals("main") || m == null;
```

Comment:
If I see this correctly, students cannot be asked to add a main-method as it would be removed in the solution model as implicit, right?

Reply:
You're right,  removing this. :)